src/build/env.py

```diff
@@ -182,9 +182,27 @@ def _create_isolated_env_venv(path):  # type: (str) -> Tuple[str, str]
     pip_distribution = next(iter(metadata.distributions(name='pip', path=[purelib])))
     pip_version = packaging.version.Version(pip_distribution.version)
 
-    # Currently upgrade if Pip 19.1+ not available, since Pip 19 is the first
-    # one to officially support PEP 517, and 19.1 supports manylinux1.
+    needs_pip_upgrade = False
     if pip_version < packaging.version.Version('19.1'):
+        # PEP-517 and manylinux1 was first implemented in 19.1
+        needs_pip_upgrade = True
+    elif platform.system() == 'Darwin':
+        # macOS 11+ name scheme change requires 20.3. Intel macOS 11.0 can be told to report 10.16 for backwards
+        # compatibility; but that also fixes earlier versions of pip so this is only needed for 11+.
+
+        #   TODO: remove when upgrading mypy from 0.800 or dropping Python 2
+        if sys.version_info >= (3,):
+            platform_mac_ver = platform.mac_ver()
+        else:
+            platform_mac_ver = cast(Any, platform.mac_ver)()
```

Comment:
Ironically, since Python 2 does't have `venv`, this never runs on python 2, so misses coverage. :P

Reply:
I would simply nest both `_create_isolated_env_venv` and `_find_executable_and_scripts` under an `if sys.version_info[0] > 2` and remove this kludge.

Reply:
The only reason this wasn't caught earlier is because `ignore_missing_imports` is set to true and `venv` is given a type of `Any` in Python 2.

Reply:
This won't keep us from writing a similar kludge if virtualenv needs to be supported too, but it would clean this up.